galsim/inclinedsersic.py

```diff
@@ -0,0 +1,216 @@
+
+    The Inclined Sersic surface brightness profile is characterized by four properties: its
+    Sersic index `n', its inclination angle (where 0 degrees = face-on and 90 degrees = edge-on),
+    its scale radius, and its scale height. The 3D light distribution function is:
+
+        I(R,z) = I_0 / (2h_s) * sech^2 (z/h_s) * exp[-b*(r/r_s)^{1/n}]
+
+    where z is the distance along the minor axis, R is the radial distance from the minor axis,
+    r_s is the scale radius of the disk, h_s is the scale height of the disk, and I_0 is the central
+    surface brightness of the face-on disk. The 2D light distribution function is then determined
+    from the scale height and radius here, along with the inclination angle.
+
+    In this implementation, the profile is inclined along the y-axis. This means that it will likely
+    need to be rotated in most circumstances.
+
+    At present, this profile is not enabled for photon-shooting.
+
+    The allowed range of values for the `n` parameter is 0.3 <= n <= 6.2.  An exception will be
+    thrown if you provide a value outside that range.  Below n=0.3, there are severe numerical
+    problems.  Above n=6.2, we found that the code begins to be inaccurate when sheared or
+    magnified (at the level of upcoming shear surveys), so we do not recommend extending beyond
+    this.  See Issues #325 and #450 for more details.
+
+    Sersic profile calculations take advantage of Hankel transform tables that are precomputed for a
+    given value of n when the Sersic profile is initialized.  Making additional objects with the
+    same n can therefore be many times faster than making objects with different values of n that
+    have not been used before.  Moreover, these Hankel transforms are only cached for a maximum of
+    100 different n values at a time.  For this reason, for large sets of simulations, it is worth
+    considering the use of only discrete n values rather than allowing it to vary continuously.  For
+    more details, see https://github.com/GalSim-developers/GalSim/issues/566.
+
+    Note that if you are building many Sersic profiles using truncation, the code will be more
+    efficient if the truncation is always the same multiple of `scale_radius`, since it caches
+    many calculations that depend on the ratio `trunc/scale_radius`.
+
+    A profile can be initialized using one (and only one) of two possible size parameters:
+    `scale_radius` or `half_light_radius`.  Exactly one of these two is required. Similarly,
+    at most one of `scale_height' and `scale_h_over_r' is required; if neither is given, the
+    default of scale_h_over_r = 0.1 will be used.
+
+    Initialization
+    --------------
+
+    @param n                The Sersic index, n.
+    @param inclination      The inclination angle, which must be a galsim.Angle instance
+    @param scale_radius     The scale radius of the exponential disk.  Typically given in arcsec.
```

Comment:
This is probably not the best place to comment, but I am a little late to the party. I find it odd that `InclinedExponential` does not take in `half_light_radius` as a parameter where as `InclinedSersic`, `Exponential` and `Sersic` do.